server/notification-providers/wecom.js

```diff
@@ -28,24 +31,76 @@ class WeCom extends NotificationProvider {
     /**
      * Generate the message to send
      * @param {object} heartbeatJSON Heartbeat details (For Up/Down only)
+     * @param {object} monitorJSON Monitor details
      * @param {string} msg General message
      * @returns {object} Message
      */
-    composeMessage(heartbeatJSON, msg) {
-        let title = "UptimeKuma Message";
-        if (msg != null && heartbeatJSON != null && heartbeatJSON["status"] === UP) {
-            title = "UptimeKuma Monitor Up";
-        }
-        if (msg != null && heartbeatJSON != null && heartbeatJSON["status"] === DOWN) {
-            title = "UptimeKuma Monitor Down";
+    composeMessage(heartbeatJSON, monitorJSON, msg) {
+        if (heartbeatJSON != null) {
+            const templateCard = {
+                card_type: "text_notice",
+                main_title: {
+                    title: this.statusToString(
+                        heartbeatJSON["status"],
+                        monitorJSON["name"]
+                    ),
+                },
+                sub_title_text: heartbeatJSON["msg"],
+                horizontal_content_list: [
+                    {
+                        keyname: "Timezone",
+                        value: heartbeatJSON["timezone"],
+                    },
+                    {
+                        keyname: "Time",
+                        value: heartbeatJSON["localDateTime"],
+                    },
+                ],
+                card_action: {
+                    type: 1,
+                    url: monitorJSON["url"]
+                        ? monitorJSON["url"]
+                        : "https://github.com/louislam/uptime-kuma", // both card_action and card_action.url are mandatory
+                },
+            };
+            if (monitorJSON["url"]) {
+                templateCard["jump_list"] = [
+                    {
+                        type: 1,
+                        url: monitorJSON["url"],
```

Comment:
In the meantime, while the other PR was dormant, we noticed that there are ways in which the url should be handled.

Instead of using `monitorJSON["url"]`, could you extract the adress like this
```js
const address = this.extractAddress(monitorJSON)
```

and then only check adress?

Example in another notification provider:

https://github.com/louislam/uptime-kuma/blob/0254e72177382dff5e02b9632abf9ef1aa48ca45/server/notification-providers/slack.js#L51